addons/account/models/account.py

```diff
@@ -975,6 +976,39 @@ def action_configure_bank_journal(self):
         # We simply call the setup bar function.
         return self.env['res.company'].setting_init_bank_account_action()
 
+    @api.multi
+    def create_invoice_from_attachment(self, attachment_ids=[]):
+        ''' Create the invoices from files.
+         :return: A action redirecting to account.invoice tree/form view.
+        '''
+        attachments = self.env['ir.attachment'].browse(attachment_ids)
+        if not attachments:
+            raise UserError(_("No attachment was provided"))
+
+        invoices = self.env['account.invoice']
+        journal_id = self or self.env['account.journal'].search([('type', '=', self.env.context.get('journal_type'))], limit=1)
+        for attachment in attachments:
+            invoice_form = Form(self.env['account.invoice'].with_context(default_journal_id=journal_id.id), view='account.invoice_supplier_form')
+            invoice = invoice_form.save()
+            attachment.write({'res_model': 'account.invoice', 'res_id': invoice.id})
+            invoice.message_post(attachment_ids=[attachment.id])
+            invoices += invoice
+
+        form_view = self.env.context.get('journal_type') == 'purchase' and self.env.ref('account.invoice_supplier_form').id or self.env.ref('account.invoice_form').id
```

Comment:
you can write the two following lines like that (following my previous comment) :
form_view = journal_type == 'purchase' and .... or ....
tree_view = journal_type == 'purchase' and .... or ....